Port fixed grids tagging

julianakwan opened the pull request:
This ports the fixed tagging criterion in which the inner L/4 of the box is tagged for refinement to be compatible with AMReX. This is useful for weak scaling tests and also examples in which chi is not needed.

julianakwan committed:
Tagging: Port fixed grid tagging

Also change from chi tagging to fixed grid tagging in BinaryBH
example

BinaryBH: Remove fill patch

Fill patch not needed for fixed grid tagging

julianakwan committed:
Tagging: Address Linter report

Also reverts back to original chi tagging criterion for BinaryBHLevel
because I had changed it to fixed grid tagging for the weak scaling
tests. I guess it should be chi tagging by default.

mirenradia commented on Source/TaggingCriteria/FixedGridsTaggingCriterion.hpp:
```diff
@@ -14,33 +14,39 @@
+    template <class data_t>
+    AMREX_GPU_DEVICE data_t
+    compute(int i, int j, int k,
+            const amrex::Array4<const data_t> &current_cell) const
```
I don't think the `current_cell` argument is used so it should be removed.
```suggestion
    compute(int i, int j, int k) const
```

julianakwan replied:
Thanks! I've removed `current_cell' as suggested.

mirenradia reviewed (changes requested):
Thanks for the PR. Just 1 minor change.

I was debating whether it's worth replacing the `simd_xxx` functions with more conventional ones but this can be done later if and when we decide to remove the SIMD infrastructure.

julianakwan committed:
Tagging: Remove current cell from function signature

As per Miren's review of PR#85, I've removed passing in the current cell
to the compute function as it is not needed in the calculations.

mirenradia reviewed (approved):
LGTM
